public/react-services/pattern-handler.js

```diff
@@ -22,80 +22,92 @@ export class PatternHandler {
-            toastLifeTimeMs: 5000
-          });
+            if (await SavedObject.getExistingIndexPattern(pattern)) {
+              await SavedObject.refreshIndexPattern(pattern);
+              getToasts().addSuccess(`${pattern} index pattern updated successfully`);
+            } else {
+              await SavedObject.createWazuhIndexPattern(pattern);
+              getToasts().addSuccess(`${pattern} index pattern created successfully`);
+            }
 
-          await SavedObject.createWazuhIndexPattern(pattern);
-          getToasts().addSuccess(`${pattern} index pattern created successfully`)
-          getDataPlugin().indexPatterns.setDefault(pattern, true);
-        } catch (err) {
-          getToasts().add({
-            color: 'error',
-            title: 'Error creating the index pattern.',
-            text: err.message || err,
-            toastLifeTimeMs: 3000
-          });
-          AppState.removeCurrentPattern();
+            getDataPlugin().indexPatterns.setDefault(pattern, true);
+          } catch (err) {
+            getToasts().add({
+              color: 'error',
+              title: 'Error creating the index pattern.',
+              text: err.message || err,
+              toastLifeTimeMs: 3000,
+            });
+            AppState.removeCurrentPattern();
 
-          this.wzMisc = new WzMisc();
-          this.wzMisc.setBlankScr(
-            'Sorry but no valid index patterns were found and creation was unsuccessful'
-          );
+            this.wzMisc = new WzMisc();
+            this.wzMisc.setBlankScr(
+              'Sorry but no valid index patterns were found and creation was unsuccessful'
+            );
+            if (
+              !window.location.hash.includes('#/settings') &&
+              !window.location.hash.includes('#/blank-screen')
+            ) {
+              window.location.href = getHttp().basePath.prepend('/app/wazuh#/blank-screen/');
+            }
+            return;
+          }
+          // retry again with the newly created index pattern
           if (
             !window.location.hash.includes('#/settings') &&
-            !window.location.hash.includes('#/blank-screen')
+            !window.location.hash.includes('#/health-check')
           ) {
-            window.location.href = '/app/wazuh#/blank-screen/';
+            window.location.href = getHttp().basePath.prepend('/app/wazuh#/health-check/');
           }
-          return;
+          patternList = await SavedObject.getListOfWazuhValidIndexPatterns(defaultPatterns, where);
         }
-        // retry again with the newly created index pattern
-        if (
-          !window.location.hash.includes('#/settings') &&
-          !window.location.hash.includes('#/health-check')
-        ) {
-          window.location.href = '/app/wazuh#/health-check/';
-        }
-        patternList = await SavedObject.getListOfWazuhValidIndexPatterns();
       }
+
```

Comment:
`return` at the end of this method, it is ok?

Reply:
thx!